### ISSUE TITLE
enum_variant_names lint gives invalid suggestions

### ISSUE THREAD

**Issue by mrmonday** [rust-lang/rust-clippy#739]
Given:

``` rust
pub enum NetworkLayer {
    Layer2,
    Layer3,
}
```

Clippy gives the following warning/help:

```
warning: All variants have the same prefix: `Layer`, #[warn(enum_variant_names)] on by default
help: remove the prefixes and use full paths to the variants instead of glob imports
```

Here, removing the prefix would lead to invalid identifiers.

**Comment by Manishearth** [comment]
Oh, that's a good point. @oli-obk want to fix this? (or I can, pretty simple)

**Comment by oli-obk** [comment]
Well. The hint should be changed, but i still consider the names to be bad. Should rather be the name of the layer instead of the number

**Comment by Manishearth** [comment]
Yeah, in this case the names could be improved (but in the general case, that's not always true)

**Comment by mrmonday** [comment]
(For the record, I plan to completely remove the version of that struct I'm actually using, the poor names are the least of the problems with it...)

I agree about the general case, which is the main reason I reported it.

**Comment by oli-obk** [comment]
This is fixed.

**Comment by CAD97** [comment]
[This came back.](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f822cafb79d77fc2a5fa4e38f42b8e3a)

```
enum Peek {
    Peek1,
    Peek2,
    Peek3,
}
```

```
warning: Variant name starts with the enum's name
 --> src/lib.rs:2:5
  |
2 |     Peek1,
  |     ^^^^^
  |
  = note: #[warn(clippy::enum_variant_names)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names

warning: Variant name starts with the enum's name
 --> src/lib.rs:3:5
  |
3 |     Peek2,
  |     ^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names

warning: Variant name starts with the enum's name
 --> src/lib.rs:4:5
  |
4 |     Peek3,
  |     ^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names

warning: All variants have the same prefix: `Peek`
 --> src/lib.rs:1:1
  |
1 | / enum Peek {
2 | |     Peek1,
3 | |     Peek2,
4 | |     Peek3,
5 | | }
  | |_^
  |
  = help: remove the prefixes and use full paths to the variants instead of glob imports
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names
```

**Comment by phansch** [comment]
It looks like this only happens when it's a private enum. Once it's made `pub`, the warning disappears.

**Comment by CAD97** [comment]
That would be because the `pub` lint is [a different lint](https://rust-lang.github.io/rust-clippy/master/#pub_enum_variant_names) in the pedantic category which is allow-by-default.
